src/units/abilities.cpp

```diff
@@ -1043,6 +1043,10 @@ bool attack_type::special_active(const config& special, AFFECTS whom,
 		(other->get_state("unpoisonable") || other->get_state(unit::STATE_POISONED))) {
 		return false;
 	}
+	if (special["id"] == "firststrike" && !is_attacker_ && other_attack_ &&
+		other_attack_->get_special_bool("firststrike", true)) {
```

Comment:
Yes, there's just one attacker and one defender, and there won't be infinite mutual recursion (checking attacker's firststrike checks defender's firststrike checks attacker's firststrike...) because of the `!is_attacker_` condition.

Your force push didn't change true to false, it pushed a different commit with exactly the same diff. I'd be happy to merge with the true changed to false (or omitted entirely), though I'd wait a day or two first to see if anyone else has comments, as I too am relatively new to the codebase.

Reply:
Ah, my bad, forgot to add the file :)

Reply:
Thanks. Approved, milestoned, will merge if no other comments.